### PR TITLE
feat: #2604 added rabbitmq listener with asyncapi validator

### PR DESCRIPTION
**Pull request by Chlenix**

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -19,7 +19,11 @@
     "noFallthroughCasesInSwitch": true,
     "resolveJsonModule": true,
     "moduleResolution": "node",
-    "typeRoots": ["node_modules/@types", "src/types/*"],
+    "baseUrl": "./",
+    "paths": {
+      "*": ["src/types/*"]
+    },
+    "typeRoots": ["node_modules/@types"],
```
Comment (azaxarov) [inline tsconfig.json]: Why the change of heart? Seems more confusing

File: src/rabbitmq/rabbitmq.ts [hunk 2]
```diff
@@ -33,7 +43,38 @@ export const rabbitConnect = (): void => {
...
         }
         ch.assertExchange(exchange, 'topic', { durable: false });
-        channel = ch;
+
+        ch.assertQueue(
+          '',
+          {
+            exclusive: true
+          },
+          (err2, q) => {
```
Comment (azaxarov) [inline src/rabbitmq/rabbitmq.ts]: Why the cryptic names?

File: tsconfig.json (hunk 1 above)
Comment (Chlenix) [reply tsconfig.json]: npm could not find my types with typeRoots

File: src/rabbitmq/rabbitmq.ts (hunk 2 above)
Comment (Chlenix) [reply src/rabbitmq/rabbitmq.ts]: I'm sorry I need to reevaluate my life decisions here
